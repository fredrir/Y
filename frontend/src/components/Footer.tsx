const Footer = () => {
  return (
    <footer className="border-t border-gray-200 bg-gray-950/80 dark:border-gray-700">
      <div className="mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8">
        <div className="flex flex-col items-center justify-between sm:flex-row">
          <div className="mb-4 flex items-center sm:mb-0">
            <span className="text-2xl font-bold text-white">Y</span>
          </div>
          <p className="text-sm text-gray-200">
            &copy; {new Date().getFullYear()} Y Corporation. All rights
            reserved.
          </p>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
