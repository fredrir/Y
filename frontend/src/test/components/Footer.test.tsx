import Footer from "@/components/Footer";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

describe("Footer", () => {
  it("renders the footer with the correct copyright year", () => {
    const { container } = render(<Footer />);
    const copyrightText = container.querySelector("p");

    expect(copyrightText).toBeInTheDocument();
    expect(copyrightText?.textContent).toContain(
      new Date().getFullYear().toString(),
    );
  });

  it("renders correctly in the DOM", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});
