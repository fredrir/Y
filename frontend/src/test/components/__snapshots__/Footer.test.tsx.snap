// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > renders correctly in the DOM 1`] = `
<DocumentFragment>
  <footer
    class="border-t border-gray-200 bg-gray-200 dark:border-gray-700 dark:bg-gray-950"
  >
    <div
      class="mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8"
    >
      <div
        class="flex flex-col items-center justify-end sm:flex-row"
      >
        <p
          class="text-sm text-gray-700 dark:text-gray-300"
        >
          Â© 2024 Y Corporation. All rights reserved.
        </p>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
